Source/ASVideoPlayerNode.mm

```diff
@@ -315,14 +321,22 @@ - (NSArray *)createDefaultControlElementArray
 
 - (void)removeControls
 {
-  for (ASDisplayNode *node in [_cachedControls objectEnumerator]) {
-    [node removeFromSupernode];
+  ASLockScope(self);
+  
+  // Grab the cached controls for removing it
+  NSMutableDictionary *cachedControls = [_cachedControls copy];
+
+  {
+    ASUnlockScope(self);
+    for (ASDisplayNode *node in [cachedControls objectEnumerator]) {
+      [node removeFromSupernode];
+    }
   }
 
-  [self cleanCachedControls];
+  [self __locked_cleanCachedControls];
 }
 
-- (void)cleanCachedControls
+- (void)__locked_cleanCachedControls
```

Comment:
ha!

Reply:
Also, should we use 1 underscore at the beginning of this method's name (i.e `-_locked_cleanCachedControls`)?